olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.h

```diff
@@ -60,6 +61,12 @@ class PartitionsRepository {
   static model::Partition PartitionFromSubQuad(
       const model::SubQuad& sub_quad, const std::string& partition);
 
+  static PartitionsResponse GetAggregatedPartitionForVersionedTile(
```

Comment:
Why do we need only aggregated partition? We can use the same for Prefetch and normal GetData(TileRequest).
Please make a more generic solution that we can reuse for other functions that do not need the nearest parent but the tile itself.

Also `GetAggregatedPartitionForVersionedTile` is not the best name. Change to `GetAggregatedTile` and make the version an optional so that we can also use it for volatile layers.